### ISSUE TITLE
maestral-gui does not shut down or hangs at shutdown

### ISSUE THREAD

**Issue by idnovic** [SamSchott/maestral-dropbox#11]
I opened maestral-gui in terminal and closed it via the tray menu item "Quit Maestral" but maestral does not shutdown without timeout. 

Few seconds after I quit maestral:
```
Shutting down threads.
Observer thread stopped.
Upload thread stopped.
```

Around a minute later
```
Shutting down threads.
Observer thread stopped.
Upload thread stopped.
Up to date
Download thread stopped.
Connection thread stopped.
```

If the sync state did not change what is the reason that maestral needs time to shutdown?

**Comment by idnovic** [comment]
I think it was a network hiccup responsible. Gonna post more after I retired it a few times.

**Comment by idnovic** [comment]
So the first test the connection state was "connecting..." because I lost wifi and just had ethernet left.

The other test were with connection state "up to date"

Still this big timeout. I do not get any logs other than the above posted. It feels like an timeout of the connection.

I measured it: 2 min and around 5 seconds.

**Comment by SamSchott** [comment]
Oups, I know where that comes from: Maestral uses the Dropbox API to query for remote changes. This query takes place in the "Download thread" and is set to timeout after 20 sec if no changes are detected. But the Dropbox API adds a random jitter of up to 90 sec of random jitter. Now, if the download thread receives the signal to stop, it won't process it until after the query returns.

I will have to change this behavior...

**Comment by idnovic** [comment]
Hey can I add my own dropbox api keys?
Or does it make sense to use m own api keys to improve this?

**Comment by SamSchott** [comment]
Right, this should be fixed with b50620db1b525b706dfd99377f7966a1c0eab897 and 996655aa08d7235434c8386c127f934c66f3e07b.

**Comment by SamSchott** [comment]
> Hey can I add my own dropbox api keys?
> Or does it make sense to use my own api keys to improve this?

I don't think this will change anything - it will mostly confuse others when linking their account. But you can of course do this for your private copy. Also, I am happy to look at pull requests!
